### ISSUE TITLE
Remove-Module PSReadLine makes input very slow on Linux

### ISSUE THREAD

**Issue by luizgn** [PowerShell/PowerShell#7604]
Steps to reproduce
------------------

```powershell
Remove-Module PSReadLine
```

Expected behavior
-----------------

After that, immediately see typed characters without colors, character positioning and other control characters.

Actual behavior
---------------

On CentOS, after that, each typed character takes about one second to be showed. 

On Ubuntu, after that, each typed character takes about one second to be showed and control characters like '33;19R1R' are inserted on input.
```powershell
PS /root> remove-module psreadline
33;19R1RPS /root>
33
19R : The term '19R' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:4
+ 33;19R
+    ~~~
+ CategoryInfo          : ObjectNotFound: (19R:String) [], CommandNotFoundException
+ FullyQualifiedErrorId : CommandNotFoundException
```

On Windows, it works fine.

Environment data
----------------

CentOS Linux release 7.5.1804 (Core)

```powershell
Name                           Value
----                           -----
PSVersion                      6.0.4
PSEdition                      Core
GitCommitId                    v6.0.4
OS                             Linux 3.10.0-862.6.3.el7.x86_64 #1 SMP Tue Jun 26 16:32:21 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Ubuntu 16.04.5 LTS

```powershell
Name                           Value
----                           -----
PSVersion                      6.0.4
PSEdition                      Core
GitCommitId                    v6.0.4
OS                             Linux 4.4.0-133-generic #159-Ubuntu SMP Fri Aug 10 07:31:43 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Windows 10
```powershell
Name                           Value
----                           -----
PSVersion                      6.0.4
PSEdition                      Core
GitCommitId                    v6.0.4
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
What terminal do you use?

**Comment by luizgn** [comment]
I log on Linux using Putty and executed pwsh, so, $env:TERM is 'xterm'.

I just tried to use 'dumb' and also tried remove $env:TERM because I noticed this variable is not set on Windows but behavior was the same.

**Comment by MaximoTrinidad** [comment]
@iSazonov,

I use their standard terminal.  This is not using ssh remoting!
:)

**Comment by iSazonov** [comment]
@luizgn Do you connect to Linux from Windows?

Please test with resently published 6.1.0 RC1 version.

**Comment by luizgn** [comment]
@iSazonov, yes, I'm connecting to Linux from Windows 10 using [Putty ](https://www.putty.org/) ssh client. 6.1.0 RC1 has same behavior.

Now, I tested using Enter-PSSession with 6.1.0 RC1 and OpenSSH_for_Windows_7.7p1 ssh client and it crashed:

```powershell
PS D:\OpenSSH-Win64> Enter-PSSession -HostName 10.0.0.2 -UserName psuser
psuser@10.0.0.2's password:
psuser@10.0.0.2's password: Enter-PSSession : The background process reported an error with the following message: The SSH client session has ended with error message: Permission denied, please try again..
At line:1 char:1
+ Enter-PSSession -HostName 10.0.0.2 -UserName psuser
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ResourceUnavailable: (:) [Enter-PSSession], PSRemotingTransportException
+ FullyQualifiedErrorId : System.Management.Automation.Remoting.PSRemotingDataStructureException,Microsoft.PowerShell.Commands.EnterPSSessionCommand
```

Than, I tested using ssh client (OpenSSH_for_Windows_7.7p1) from powershell 6.1.0 RC1 console and it has same behavior also (psuser uses /bin/pwsh-preview as shell):

```powershell
PS D:\OpenSSH-Win64> ssh psuser@10.0.0.2
psuser@10.0.0.2's password:
PowerShell 6.1.0-rc.1
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/pscore6-docs
Type 'help' to get help.

PS /home/psuser> $psversiontable

Name                           Value
----                           -----
PSVersion                      6.1.0-rc.1
PSEdition                      Core
GitCommitId                    6.1.0-rc.1
OS                             Linux 3.10.0-862.6.3.el7.x86_64 #1 SMP Tue Jun 26 16:32:21 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


PS /home/psuser> remove-module psreadline
PS /home/psuser> test
```

After removing psreadline, all characters are typed very slowly.

**Comment by MaximoTrinidad** [comment]
I don't think the issue is with remoting.  It seems this is at the Linux main console terminal.

Also, I had a different issue when removing PSReadline impacting tab-completion which ended up to be **By-Design**.  

In other words, in Linux or MacOS, we shouldn't remove the PSReadLine module. There's a chance it will impact the performance of PowerShell Core.

:(

**Comment by lzybkr** [comment]
PSReadLine is not used for prompts, e.g. `$host.UI.ReadLine()` or `[Console]::ReadLine()`, so it's worth understanding if that's also slow.

As an aside, `$host.UI.ReadLine()` should probably use PSReadLine, but disable colors and history.

**Comment by PaulHigin** [comment]
I agree that this is not about PowerShell SSH based remoting.  This seems to only repro when SSH remoting from Windows to Linux, so we should make the Win32-OpenSSH team aware.  I have not been able to repro this but I am currently using an older version of Windows OpenSSH.

@manojampalam Could this be something related to Win32-OpenSSH?

**Comment by luizgn** [comment]
@MaximoTrinidad could you provide more detail about your environment? All my tries caused a high latency on text input.

And, for me, it doesn't happen with Win32-OpenSSH only, it also happens with putty.

**Comment by manojampalam** [comment]
Unable to comprehend this thread in one pass through :(. If you feel there is an issue with Win32-OpenSSH, please open an issue here - https://github.com/PowerShell/Win32-OpenSSH/issues

**Comment by MaximoTrinidad** [comment]
@luizgn,

Please clarify if this issues is due to using PSRemoting or just SSH. The issue is not clear.  Also, provide the exact steps to reproduce as the steps provided are not sufficient.  

What happen before removing PSReadLine?

:)

**Comment by luizgn** [comment]
Hi, I tested using ssh, putty and PSRemoting. Issues happens in all, so, I'm sure it is not a Win32-OpenSSH issue.

**Comment by MaximoTrinidad** [comment]
OK! So, this issue is really while doing remoting and not at each individual OS consoles.

Let me check it out now that I understand the environment and condition.  

:)

**Comment by luizgn** [comment]
@maximoTrinidad, @iSazonov, do you need more information?

**Comment by MaximoTrinidad** [comment]
Well,  I can't test!

When connecting Windows to either Ubuntu 17.10 or to CentOS 7 using Enter-PSSession, it won't let me remove the module PSReadLine giving the following message:

```
[maxt@mars]: PS /home/maxt> remove-module psreadline
remove-module : No modules were removed. Verify that the specification of modules to remove is correct and those modules exist in the runspace.
+ CategoryInfo          : ResourceUnavailable: (:) [Remove-Module], InvalidOperationException
+ FullyQualifiedErrorId : Modules_NoModulesRemoved,Microsoft.PowerShell.Commands.RemoveModuleCommand
```

The module is installed but can't be remove.

I only can remove it in Windows OS in PowerShell Core.  So, something change in Windows 10.

I'm on Windows 10 Insider Build 17746_rs5.  Also, it seems at now Windows 10 latest Insider build has OpenSSH integrated behind the scene. I can't find any active ssh services any more on the Services Panel.

**Comment by SteveL-MSFT** [comment]
PSCore6 console host doesn't write any explicit VT100 sequences, that `33;19R` sequence is a response from the terminal on a cursor position request.  This is being sent by .NET CoreFx console api and is supposed to be handled by the api so you don't see that output.  One significant difference between Windows PowerShell and PSCore6 is that the PSCore6 console host has a very simplistic readline implementation and PSReadLine is expected to be loaded for the interactive experience.  @luizgn is there a reason you need to remove PSReadLine?

**Comment by luizgn** [comment]
Hi @SteveL-MSFT, sorry delay to answer.

I have an application which starts Windows PowerShell and interactively send commands to it and get responses. Linux support is amazing for me because I will be able to use this on both platforms. However, I'm receiving control characters PSCore6. I think PSReadLine is causing that, so, I would like to remove it.. I appreciate if there is other solution to avoid control characters.

This is an example of control characters executing command 'hostname':

![image](https://user-images.githubusercontent.com/17809978/47298988-84feef00-d5ef-11e8-825b-ff79861e35ce.png)

By the way, this is a short video showing what happens when I remove PSReadLine on Linux. All characters typed after that are showed very slowly:

[powershell2.zip](https://github.com/PowerShell/PowerShell/files/2501851/powershell2.zip)

On Windows, I can remove PSReadLine without problems.

**Comment by SteveL-MSFT** [comment]
@luizgn I believe the perf issue is a dupe of https://github.com/PowerShell/PowerShell/issues/7971

If your application is interacting with PowerShell, you should consider hosting powershell rather than interacting with pwsh via stdin/stdout.
